code/modules/mob/living/living_movement.dm

```diff
@@ -56,11 +60,137 @@
 			return
 	remove_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING)
 
-/mob/living/canZMove(dir, turf/target)
-	return can_zTravel(target, dir) && (movement_type & FLYING)
+#define MOVETYPE_NONE_JUMP -1
+#define MOVETYPE_NONE 0
+#define MOVETYPE_CLIMB 1
+#define MOVETYPE_FLY 2
+#define MOVETYPE_JETPACK 3
+#define MOVETYPE_FLOAT 4
+#define MOVETYPE_JAUNT 5
+
+/mob/living/canZMove(dir, turf/source, turf/target, pre_move = TRUE)
+	if(incapacitated(check_immobilized = TRUE) || resting || zmoving || IsKnockdown())
+		return MOVETYPE_NONE
+	if(incorporeal_move || (movement_type & PHASING))
+		return MOVETYPE_JAUNT
+	var/upwards = dir == UP
+	if(((upwards && !target.allow_z_travel) || (!upwards && !source.allow_z_travel)))
+		return MOVETYPE_NONE
+	var/can_climb = turf_can_climb(upwards ? target : source)
```

Comment:
If we are going to have climbing, it should either be for outdoor turfs like climbing the sides of mountains, or a trait that isn't available to most station humans (so outside mobs and stuff like xenos can climb). I don't think regular crew should be able to climb over walls and just jump up onto lattices.

Reply:
Climbing only applies for lattices. I think it's a pretty useful mechanic as ztravel is usually impossible for most crew and they act as a slower, cheap alternative to ladders. It also takes 3x longer than flight and you can't move at all during the climb (unlike flight)

Reply:
Indoor climbing should not exist at all.

Reply:
Climbing removed entirely (including lattices), but the code will remain in if it is ever modified or improved somehow.